core/src/main/kotlin/com/strumenta/kolasu/serialization/JsonGenerator.kt

```diff
@@ -130,27 +131,28 @@ class JsonGenerator {
         if (result.root == null) {
             writer.nullValue()
         } else {
-            result.root.toJsonStreaming(writer, shortClassNames)
+            generateJSONWithStreaming(result.root, writer, shortClassNames)
         }
         writer.endObject()
     }
 
     fun generateJSONWithStreaming(root: Node, writer: JsonWriter, shortClassNames: Boolean = false) {
-        root.toJsonStreaming(writer, shortClassNames)
+        val gson = gsonBuilder.setPrettyPrinting().create()
+        gson.toJson(generateJSON(root, null, null, null, shortClassNames), writer)
```

Comment:
here we could specify the names of the parameters, to make this more understandable